### ISSUE TITLE
Disable use of "rancher-metadata" in microservices and use the IP (169.254.169.250) instead

### ISSUE THREAD

**Issue by leodotcloud** [rancher/rancher#9521]
There is no point in making the microservices generate weird DNS requests and making things complicated. Instead we should just use the IP address directly to make things simpler.
We can make this IP configurable from the arguments instead of hardcoding to provide flexibility to change.

The following requests are from `rancher-kubernetes-agent` container.
```
01:05:17.172496 IP (tos 0x0, ttl 63, id 19987, offset 0, flags [DF], proto UDP (17), length 90)
    10.42.123.78.52548 > 169.254.169.250.53: [bad udp cksum 0xd9c8 -> 0x9653!] 56295+ AAAA? rancher-metadata.kubernetes.rancher.internal. (62)
        0x0000:  02e8 8fe4 5307 0242 c590 b669 0800 4500  ....S..B...i..E.
        0x0010:  005a 4e13 4000 3f11 140f 0a2a 7b4e a9fe  .ZN.@.?....*{N..
        0x0020:  a9fa cd44 0035 0046 d9c8 dbe7 0100 0001  ...D.5.F........
        0x0030:  0000 0000 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 6b75 6265 726e 6574  etadata.kubernet
        0x0050:  6573 0772 616e 6368 6572 0869 6e74 6572  es.rancher.inter
        0x0060:  6e61 6c00 001c 0001                      nal.....
01:05:17.172583 IP (tos 0x0, ttl 63, id 19988, offset 0, flags [DF], proto UDP (17), length 90)
    10.42.123.78.39041 > 169.254.169.250.53: [bad udp cksum 0xd9c8 -> 0x758f!] 12682+ A? rancher-metadata.kubernetes.rancher.internal. (62)
        0x0000:  02e8 8fe4 5307 0242 c590 b669 0800 4500  ....S..B...i..E.
        0x0010:  005a 4e14 4000 3f11 140e 0a2a 7b4e a9fe  .ZN.@.?....*{N..
        0x0020:  a9fa 9881 0035 0046 d9c8 318a 0100 0001  .....5.F..1.....
        0x0030:  0000 0000 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 6b75 6265 726e 6574  etadata.kubernet
        0x0050:  6573 0772 616e 6368 6572 0869 6e74 6572  es.rancher.inter
        0x0060:  6e61 6c00 0001 0001                      nal.....
01:05:17.172951 IP (tos 0x0, ttl 64, id 52314, offset 0, flags [DF], proto UDP (17), length 126)
    169.254.169.250.53 > 10.42.123.78.39041: [bad udp cksum 0xd9ec -> 0xc598!] 12682 NXDomain*- q: A? rancher-metadata.kubernetes.rancher.internal. 0/1/0 ns: rancher.internal. [1s] SOA rancher.internal. rancher.internal. 39239 60 10 86400 1 (98)
        0x0000:  0242 c590 b669 02e8 8fe4 5307 0800 4500  .B...i....S...E.
        0x0010:  007e cc5a 4000 4011 94a3 a9fe a9fa 0a2a  .~.Z@.@........*
        0x0020:  7b4e 0035 9881 006a d9ec 318a 8503 0001  {N.5...j..1.....
        0x0030:  0000 0001 0000 1072 616e 6368 6572 2d6d  .......rancher-m
        0x0040:  6574 6164 6174 610a 6b75 6265 726e 6574  etadata.kubernet
        0x0050:  6573 0772 616e 6368 6572 0869 6e74 6572  es.rancher.inter
        0x0060:  6e61 6c00 0001 0001 c028 0006 0001 0000  nal......(......
        0x0070:  0001 0018 c028 c028 0000 9947 0000 003c  .....(.(...G...<
        0x0080:  0000 000a 0001 5180 0000 0001            ......Q.....
```

**Comment by leodotcloud** [comment]
Fixed in `ipsec: 0.1.4` with image: `rancher/net:v0.11.8`.
Fixed in `vxlan: v0.2.1` with image: `rancher/net:v0.11.8`.

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.6.8-rc4 using the following infra images:

Ipsec router with rancher/net:v0.11.8 

vxlan router with rancher/net:v0.11.8
rancher-kubernetes-agent with rancher/kubernetes-agent:v0.6.5
rancher-kubernetes-auth with rancher/kubernetes-auth:v0.0.8


K8s and cattle validation tests did not report any new issues.
